src/main/java/uk/gov/pay/api/model/links/PaymentWithAllLinks.java

```diff
@@ -100,4 +101,201 @@ public static PaymentWithAllLinks getPaymentWithLinks(
+        }
+
+        public PaymentWithAllLinksBuilder setPaymentConnectorResponseLinks(List<PaymentConnectorResponseLink> paymentConnectorResponseLinks) {
+            this.paymentConnectorResponseLinks = paymentConnectorResponseLinks;
+            return this;
+        }
+
+        public PaymentWithAllLinksBuilder setSelfLink(URI selfLink) {
+            this.selfLink = selfLink;
+            return this;
+        }
+
+        public PaymentWithAllLinksBuilder setPaymentEventsUri(URI paymentEventsUri) {
+            this.paymentEventsUri = paymentEventsUri;
+            return this;
+        }
+
+        public PaymentWithAllLinksBuilder setPaymentCancelUri(URI paymentCancelUri) {
+            this.paymentCancelUri = paymentCancelUri;
+            return this;
+        }
+
+        public PaymentWithAllLinksBuilder setPaymentRefundsUri(URI paymentRefundsUri) {
+            this.paymentRefundsUri = paymentRefundsUri;
+            return this;
+        }
+
+        public PaymentWithAllLinksBuilder setPaymentCaptureUri(URI paymentCaptureUri) {
+            this.paymentCaptureUri = paymentCaptureUri;
+            return this;
+        }
+
+        public PaymentWithAllLinksBuilder setPaymentAuthorisationUri(URI paymentAuthorisationUri) {
+            this.paymentAuthorisationUri = paymentAuthorisationUri;
+            return this;
+        }
+
+        public PaymentWithAllLinksBuilder setCorporateCardSurcharge(Long corporateCardSurcharge) {
+            this.corporateCardSurcharge = corporateCardSurcharge;
+            return this;
+        }
+
+        public PaymentWithAllLinksBuilder setTotalAmount(Long totalAmount) {
+            this.totalAmount = totalAmount;
+            return this;
+        }
+
+        public PaymentWithAllLinksBuilder setProviderId(String providerId) {
+            this.providerId = providerId;
+            return this;
+        }
+
+        public PaymentWithAllLinksBuilder setMetadata(ExternalMetadata metadata) {
+            this.metadata = metadata;
+            return this;
+        }
+
+        public PaymentWithAllLinksBuilder setFee(Long fee) {
+            this.fee = fee;
+            return this;
+        }
+
+        public PaymentWithAllLinksBuilder setNetAmount(Long netAmount) {
+            this.netAmount = netAmount;
+            return this;
+        }
+
+        public PaymentWithAllLinksBuilder setAuthorisationSummary(AuthorisationSummary authorisationSummary) {
+            this.authorisationSummary = authorisationSummary;
+            return this;
+        }
+
+        public PaymentWithAllLinksBuilder setAgreementId(String agreementId) {
+            this.agreementId = agreementId;
+            return this;
+        }
+
+        public PaymentWithAllLinksBuilder setAuthorisationMode(AuthorisationMode authorisationMode) {
+            this.authorisationMode = authorisationMode;
+            return this;
+        }
+
+        public PaymentWithAllLinks createPaymentWithAllLinks() {
```

Comment:
should this method be named `build` to be consistent with other builders?

Reply:
yeah I'll make it consistent. i had let intellij do all the auto-generation here.